client/session.go

```diff
@@ -3291,20 +3295,22 @@ func (r *bulkBlocksResult) addBlockFromPeer(id ident.ID, peer topology.Host, blo
 
 		// If there are no data in the current block, there is no
 		// need to merge
-		if currReader == nil {
+		if currReader.IsEmpty() {
 			continue
 		}
 
 		resultReader, err := result.Stream(tmpCtx)
 		if err != nil {
 			return err
 		}
-		if resultReader == nil {
+		if resultReader.IsEmpty() {
 			return nil
 		}
 
-		readers := []io.Reader{currReader, resultReader}
-		encoder, err := r.mergeReaders(start, readers)
+		readers := []xio.SegmentReader{currReader, resultReader}
```

Comment:
To avoid interface allocation here too can we do `currReader.SegmentReader()` and `resultReader.SegmentReader()`?  Avoid all the allocs =0